entities/src/main/java/com/flipkart/varadhi/entities/TopicPartitions.java

```diff
@@ -20,7 +20,9 @@ public static <T extends StorageTopic> TopicPartitions<T> byPartitions(T topic,
     }
 
     public static <T extends StorageTopic> TopicPartitions<T> byTopic(T topic) {
-        return new TopicPartitions<>(topic, null);
+        //TODO:: Discuss this
+        // Can a topic be non partitioned or assume there is always 1 (implicit).
```

Comment:
yes, the topic can be "non-partitioned", or even if partitioned, then using this method means implicitly that you are talking about the topic as a whole (including all of its partitions).